lib/rubocop/cop/naming/method_name.rb

```diff
@@ -6,12 +6,15 @@ module Naming
       # Makes sure that all methods use the configured style,
       # snake_case or camelCase, for their names.
       #
-      # This cop has `AllowedPatterns` configuration option.
+      # This cop has `AllowedPatterns` and `ForbiddenIdentifiers` configuration options.
```

Comment:
I'd probably explain the purpose of these options a bit more. For `ForbiddenIndentifiers` it might also make sense to populate it by default with the reserved names we want to avoid.

Reply:
I don't think there are any names that should be forbidden by default, what did you have in mind?

As mentioned on https://github.com/rubocop/rubocop/pull/13620#discussion_r1897704384 many Ruby keywords are widely used as method names in different contexts.

Reply:
Fair enough. At the very least I think it should be clear what people are supposed to use this for, as I don't believe that's obvious right now, given the historical purpose of this particular cop.

Reply:
I think that makes sense? `__send__` can be added as well (but not `send` since `Socket#send` exists in stdlib). It gets a bit murky because `BasicObject` only has the underscore methods, but, the majority will be through `method_missing` and similar I believe.

When doing a code search for `__id__` I think I'm mainly getting ruby implementations https://github.com/search?q=%2Fdef+__id__%5B%5Ea-zA-Z_%5D%2F+lang%3Aruby+-is%3Afork+-path%3A*.rbi&type=code. Same for `__send__` probably.

`object_id` is similar but gives _some_ legit hits https://github.com/search?q=%2Fdef+object_id%5B%5Ea-zA-Z_%5D%2F+lang%3Aruby+-is%3Afork+-path%3A*.rbi&type=code.

I think most redefinitions for these happen accidentally through metaprogramming (like a `Object` model in Rails).

Reply:
Since `__id__` and `__send__` give actual ruby warnings, I'd kind of prefer having a specific `Lint` cop for those, but having them be forbidden by default could work too.

Reply:
Thanks for the inputs everyone! I've added `__id__` and `__send__` to default forbidden identifiers. And also added some explanation these config options to the doc. cc @bbatsov @dvandersluis